### PR TITLE
[Cofense-32] Cofense Intelligence V2 Enhancement

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Enhancement of Cofense Intelligence V2

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2_test.py [hunk 1]
```diff
@@ -11,11 +18,67 @@
...
+    {'brand': 'Cofense Intelligence', 'entityA': '127.0.0.1', 'entityAFamily': 'Indicator', 'entityAType': 'IP',
+     'entityB': 'url', 'entityBFamily': 'Indicator', 'entityBType': 'URL', 'fields': {}, 'name': 'related-to',
+     'reverseName': 'related-to', 'type': 'IndicatorToIndicator'},
+    {'brand': 'Cofense Intelligence', 'entityA': '127.0.0.1', 'entityAFamily': 'Indicator', 'entityAType': 'IP',
+     'entityB': 'md5', 'entityBFamily': 'Indicator', 'entityBType': 'File', 'fields': {}, 'name': 'related-to',
+     'reverseName': 'related-to', 'type': 'IndicatorToIndicator'},
+    {'brand': 'Cofense Intelligence',
+     'entityA': '127.0.0.1', 'entityAFamily': 'Indicator', 'entityAType': 'IP', 'entityB': 'md5',
+     'entityBFamily': 'Indicator', 'entityBType': 'File', 'fields': {}, 'name': 'related-to',
+     'reverseName': 'related-to', 'type': 'IndicatorToIndicator'}]
```
Comment (ostolero) [inline Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2_test.py:75]: after changing the create_relationship function (see comment in function regarding entityBType), change these accordingly

File: Packs/CofenseIntelligenceV2/ReleaseNotes/1_1_0.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#### Integrations
+##### Cofense Intelligence v2
+- Added 1 reputation command:
+    - ***domain***
+- Added relationship to all the reputation commands
+
```
Comment (ostolero) [inline Packs/CofenseIntelligenceV2/ReleaseNotes/1_1_0.md:6]: Add description for the changes made to 'cofense-search' command?

**Review by ostolero — COMMENTED** [review]
Hi! thank you for the contribution, overall looks great! see the comments added.
Mainly note these two comments in the integration-
1. Change the use of Global param SEVERITY_SCORE
2. Create some conversion for indictor type in 'create_relationship' function.
Feel free to contact me if you have any questions! tnx

File: Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py [hunk 3]
```diff
@@ -302,17 +404,20 @@ def search_url_command(client: Client, args: Dict[str, Any], params) -> List[Com
         dbot_score_obj = Common.DBotScore(indicator=url, indicator_type=DBotScoreType.URL,
                                           integration_name=INTEGRATION_NAME, score=dbot_score,
                                           reliability=params.get(RELIABILITY))
-        url_indicator = Common.URL(url=url, dbot_score=dbot_score_obj)
+        relationships = create_relationship(client, url, threats, FeedIndicatorType.URL)
+        url_indicator = Common.URL(url=url, dbot_score=dbot_score_obj, relationships=relationships)
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py:408]: @ostolero We have many threat reports tied to a given URL, each with its own malware family, campaign, and other information as seen in the given below image. So my concern is that we can't decide which details to keep in the URL Object fields. 
image

File: Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py [hunk 4]
```diff
@@ -470,16 +582,21 @@ def extracted_string(client: Client, args: Dict[str, Any], params) -> CommandRes
...
 
-    limit = args.get('limit')
+    limit = arg_to_number(args.get('limit'))
     if not limit:
         limit = 10
-
+    malware_family = args.get('malware_family')
+    malware_file = args.get('malware_file')
+    malware_subject = args.get('malware_subject')
+    url = args.get('url')
```
[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py:592]: To expand the command options

File: Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ostolero) [reply Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py:429]: I see, then we'll leave it as is :)

File: Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py [hunk 5]
```diff
@@ -499,8 +616,53 @@ def extracted_string(client: Client, args: Dict[str, Any], params) -> CommandRes
...
+        result = client.threat_search_call(domain=domain, days_back=days_back)
+        threats = result.get('data', {}).get('threats', [])
+        remove_false_vendors_detections_from_threat(threats)
+        outputs = {'Data': domain, 'Threats': threats}
+        md_data, dbot_score = threats_analysis(threats, indicator=domain, threshold=params.get('domain_threshold'))
+        dbot_score_obj = Common.DBotScore(indicator=domain, indicator_type=DBotScoreType.DOMAIN,
+                                          integration_name=INTEGRATION_NAME, score=dbot_score,
+                                          reliability=params.get(RELIABILITY))
+        relationships = create_relationship(client, domain, threats, FeedIndicatorType.Domain)
+        domain_indicator = Common.Domain(domain=domain, dbot_score=dbot_score_obj, relationships=relationships)
```
[the remark this answers is not in the document]
Comment (ostolero) [reply Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py:674]: Is this the same situation as in the URL indictor? (multiple campaigns and malware families?)

[the remark this answers is not in the document]
Comment (crestdatasystems) [reply Packs/CofenseIntelligenceV2/Integrations/CofenseIntelligenceV2/CofenseIntelligenceV2.py:674]: Yes @ostolero , same is the case for all reputation commands in this XSOAR pack

**Review by ostolero — APPROVED** [review]
All change request were addressed let's schedule a demo :)

**Comment by ostolero** [comment]
Hi, great demo! no changes needed, you can merge the pr.

**Comment by crestdatasystems** [comment]
Hi @ostolero thank you! Can you also remove pending-demo label and merge this PR as only those with write access to this repository will be able to merge.

**Pull request merged**